enhancements/monitoring/alerts-ui-management.md

```diff
@@ -0,0 +1,642 @@
+- `source`
+- `notification_receiver`
+- Other alert labels (arbitrary `labelKey=labelValue` pairs)
+
+**Health**
+
+| Method | Path        | Description                 | Notes |
+|--------|-------------|-----------------------------|-------|
+| GET    | `/health`   | Basic health‑check endpoint |       |
+
+
+
+### Data Model
+- Store user alerts in `PrometheusRule`s in the specified project.
+- Update platform alerts by writing relabel rules into the `AlertRelabelConfig` CR in `cluster-monitoring-config`.
+- Use labels for disable/enable flags; rely on existing CRDs - Applies only for Platform alert rules.
+
+### Migration
+- Automated tool to detect existing `PrometheusRule` CRDs and import them into the new system
+- Preserve existing labels, annotations and support rollback on errors
+
+### GitOps / Argo CD Compliance
+- All generated `PrometheusRule` and `AlertRelabelConfig` resources remain declarative and suitable for committing to Git
+
+### Workflow Description
+
+## Pain Points Addressed by this Design
+- **Generic and Non-Actionable Alerts:** The design will provide more actions from an alert and link to resources like logs and Korrel8r for troubleshooting.
+- **Alert Noise and Data Overload:** Grouping, advanced filters, and saved filters will help reduce noise and the need for repetitive filtering.
+- **Missed Alarms or Missing Data:** Users will be able to create flexible alert definitions directly in the UI to monitor any data type, configure notifications, and link a runbook.
+
+## Pain Points Not Directly Addressed
+- High Resource Consumption of Observability Tools
+- Lack of Unified Views (within a single-cluster scope)
+
+## Risks & Mitigations
+- **Performance**: split large rule sets into multiple CRs; benchmark reconciliation latency
+- **Misconfiguration**: validate syntax; prevent disabling protected base alerts via exclusion list
+- **Complexity**: sensible defaults; UI guidance; limit advanced options in MVP
+
+## Graduation Criteria
+- **Tech Preview**: basic CRUD, clone, disable/enable, silencing via UI/API; migration tool v1
+- **GA**: best‑practice guidance in UI; multi‑namespace filtering; full test coverage; complete documentation
+
+## Open Questions
+1. **Per‑Alert vs. Single‑File**: Should each user‑defined alert reside in its own `PrometheusRule` file, or group all into one? A customer noted per‑alert files may simplify GitOps/Argo CD maintenance—does that hold true at scale? - Agreed that this would be up to the user to define.
+2. **Read‑Only Detection**: Which annotations, labels or ownerReferences reliably indicate GitOps‑managed resources to render them read‑only in our UI?
+3. **Concurrent Operator Updates**: How should we handle cases where upstream operators update their own `PrometheusRule` CRs—should we reconcile `AlertRelabelConfig` entries periodically? - [simonpasquier] Replied that, as of now, the cluster admin will need to detect the drift and update their customization.
```

Comment:
correct at least for MVP. We could store a hash of the original definition, compare this to the actual definition and provide a notification to the user (e.g. "the definition of rule XYZ that you've customized has changed") but it would be the responsibility of the user to decide what they want to do (drop the override, adjust it or do nothing).